### DISCUSSION TITLE
Icons margin

### DISCUSSION THREAD

**Discussion by grolongo** [company-mode/company-mode#1361] [Q&A] [answered]
Hi! Just a quick question, is it possible to add a few pixels spaces before and after the icons to make it look less cramped like how it's done with VSCode intellisense?  
I made an image with two arrows in case I wasn't clear enough.  
Thank you!

![ex](https://user-images.githubusercontent.com/34292770/210629328-6fa2da84-73db-407b-ac8e-9d66069dce97.png)

**Comment by dgutov** [comment, marked as answer]
Hi!

First of all, you can try `(setq company-icon-margin 3)`. That should add 1 columns of spacing between the icon and the text.

Second, you can add try patch on top of that:

```diff
diff --git a/company.el b/company.el
index 79d2619..4d35f52 100644
--- a/company.el
+++ b/company.el
@@ -1634,9 +1634,11 @@ end of the match."
                                        bkg)))
              (spacer-px-width (- (* company-icon-margin dfw) icon-size)))
         (concat
+         (propertize (company-space-string 1)
+                     'display `(space . (:width (,(/ spacer-px-width 2)))))
          (propertize " " 'display spec)
-         (propertize (company-space-string (1- company-icon-margin))
-                     'display `(space . (:width (,spacer-px-width))))))
+         (propertize (company-space-string 1)
+                     'display `(space . (:width (,(/ spacer-px-width 2)))))))
     nil))
 
 (defun company-vscode-dark-icons-margin (candidate selected)
```

Apply and re-evaluate the changed function with `C-M-x`.

Do you like either of the looks?

**Reply by grolongo** [reply]
Hi @dgutov thanks a lot for the help. I should have look better into the company customize-group to find `company-icon-margin`, my bad.  

Anyway, having `(setq company-icon-margin 3)` _with_ the patch you provided at the same time does exactly what I want, it looks just perfect (in my opinion).  

Is there a drawback into having this patch merged by default into your package?  

Thanks again!

**Reply by dgutov** [reply]
It's an extra code path with a little duplication, which is not ideal, but not too much of a problem either.

I just wanted some testing and confirmation first, before pushing this out to master.

**Reply by dgutov** [reply]
Now `(setq company-icon-margin 3)` should be enough.
